### PR TITLE
Update AssignCulture Task to support existing Culture metadata

### LINKED ISSUE #9954: [Bug]: Valid BCP 47 tags in the resource file name not recognized
### Issue Description

I was trying to add support of an uncommon language to my app with resource files, before I realized its language tag was not being recognized by MSBuild, and it was not generating the expected satellite assembly for that language.

### Steps to Reproduce

1. Create a project with three resource files: `Resources.resx`, `Resources.zh.resx`, and `Resources.yue.resx`.
2. Build the project.

### Expected Behavior

The satellite assemblies for both `zh` and `yue` are generated.

### Actual Behavior

The satellite assembly for `zh` is generated, but the one for `yue` is not.

### Analysis

I suspect that the following code is causing this issue.

https://github.com/dotnet/msbuild/blob/7ca3c98fad986066bbf2802c863236b4a0f4e34a/src/Tasks/CultureInfoCache.cs#L70

A valid language code is not necessarily predefined.

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by f-alizada**
Fixes #9954

### Context
Currently there is a difference in public documentation and actual implementation of AssignCulture task. 

### The difference: 
```If the incoming item from the Files parameter already contains a Culture metadata entry, the original metadata entry is used.```

#### In fact the Culture is not respected. 

Please see the comment: https://github.com/dotnet/msbuild/issues/9954#issuecomment-2049388980

### Changes Made
Update the AssignCulture task to include the property that will allow switch the way existing Culture metadata is handled. 

- If RespectAlreadyAssignedItemCulture is set to false
   - the Culture will not be respected. 
- If RespectAlreadyAssignedItemCulture = true 
   - set Culture metadata is respected and the CultureNeutralAssignedFiles will be populated with unchanged filename of Files. 

Updated the Microsoft.Common.CurrentVersion.targets to make it possible to set the property on the project

### Testing
Added unit test + manually tested on the local project.

### CODE REVIEW

File: src/Tasks.UnitTests/AssignCulture_Tests.cs [hunk 1]
```diff
@@ -277,5 +261,47 @@ public void Pseudolocales_CaseInsensitive()
             Assert.Equal($"MyResource.{culture}.resx", t.AssignedFiles[0].ItemSpec);
             Assert.Equal("MyResource.resx", t.CultureNeutralAssignedFiles[0].ItemSpec);
         }
+
+        /// <summary>
+        /// Any pre-existing Culture attribute on the item is to be repected
```
Comment (ladipro) [inline src/Tasks.UnitTests/AssignCulture_Tests.cs:266]: nit: Typo "repected"

**Review by ladipro — APPROVED** [review]
Looks great!

File: src/Tasks/AssignCulture.cs [hunk 2]
```diff
@@ -41,6 +41,12 @@ public class AssignCulture : TaskExtension
         [Required]
         public ITaskItem[] Files { get; set; } = Array.Empty<ITaskItem>();
 
+        /// <summary>
+        /// If the flag set to 'true' the incoming list with existing Culture metadata will not be ammended and CultureNeutralAssignedFiles filename will be equal to the original.
+        /// In case the Cutlture metadata was not provided, the logic of RespectAlreadyAssignedItemCulture will not take any effect.
```
[the remark this answers is not in the document]
Comment (f-alizada) [reply src/Tasks/AssignCulture.cs]: Thank you! Updated

File: src/Tasks.UnitTests/AssignCulture_Tests.cs (hunk 1 above)
Comment (f-alizada) [reply src/Tasks.UnitTests/AssignCulture_Tests.cs]: Thank you! Addressed

**Review by surayya-MS — APPROVED** [review]

File: src/Tasks.UnitTests/AssignCulture_Tests.cs [hunk 3]
```diff
@@ -194,12 +179,11 @@ public void Regress283991()
...
 
-        /*
-        * Method:   PseudoLocalization
-        *
-        * Test the usage of Windows Pseudo-Locales
-        * https://docs.microsoft.com/en-gb/windows/desktop/Intl/pseudo-locales
-        */
+        /// <summary>
+        /// Test the usage of Windows Pseudo-Locales
+        /// * https://docs.microsoft.com/en-gb/windows/desktop/Intl/pseudo-locales
```
Comment (surayya-MS) [inline src/Tasks.UnitTests/AssignCulture_Tests.cs:184]: nit: remove * from previous multiline comment

**Review by surayya-MS — APPROVED** [review]

**Review by maridematte — APPROVED** [review]

File: src/Tasks/Microsoft.Common.CurrentVersion.targets [hunk 4]
```diff
@@ -3244,6 +3244,10 @@ Copyright (C) Microsoft Corporation. All rights reserved.
       Name="SplitResourcesByCulture"
       DependsOnTargets="AssignTargetPaths">
 
+    <PropertyGroup>
+      <RespectAlreadyAssignedItemCulture Condition="'$(RespectAlreadyAssignedItemCulture)' == ''">false</RespectAlreadyAssignedItemCulture>
```
Comment (rainersigwald) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets:3248]: Is there a rollout plan for making this `true`?

File: src/Tasks.UnitTests/AssignCulture_Tests.cs [hunk 5]
```diff
@@ -277,5 +261,47 @@ public void Pseudolocales_CaseInsensitive()
...
+            Assert.Equal("en-GB", t.AssignedFiles[0].GetMetadata("Culture"));
+            Assert.Equal("MyResource.fr.resx", t.AssignedFiles[0].ItemSpec);
+            Assert.Equal("MyResource.fr.resx", t.CultureNeutralAssignedFiles[0].ItemSpec);
+        }
+
+        /// <summary>
+        /// Any pre-existing Culture attribute on the item is not to be respected, because culture is not set
+        /// </summary>
+        [Fact]
+        public void CultureMetaDataShouldNoBeRespected()
```
Comment (rainersigwald) [inline src/Tasks.UnitTests/AssignCulture_Tests.cs:290]:
```suggestion
        public void CultureMetaDataShouldNotBeRespected()
```

**Review by rainersigwald — APPROVED** [review]

File: src/Tasks/Microsoft.Common.CurrentVersion.targets (hunk 4 above)
Comment (f-alizada) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:3248]: Yes, once this is checked in -> enabling possibility to to use the predefined Culture metada functionality. 
Within the next release I'll enable this behaviour by default (under the change wave 17.12). I planned to file a issue not to forget it (after PR is merged).

**Pull request merged**